pkg/pgmodel/model/labels.go

```diff
@@ -9,23 +9,50 @@ import (
 	"fmt"
 	"math"
 	"sort"
+	"strconv"
 	"strings"
 	"sync"
 
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/timescale/promscale/pkg/prompb"
 )
 
+// SeriesID represents a globally unique id for the series. This should be equivalent
+// to the PostgreSQL type in the series table (currently BIGINT).
+type SeriesID int64
+
+const unsetSeriesID = -1
+
+func (s SeriesID) String() string {
+	return strconv.FormatInt(int64(s), 10)
+}
+
+//Epoch represents the series epoch
+type SeriesEpoch int64
```

Comment:
Sorry, but whenever I read the epoch thing, I still get confused even today. Can we please add a good explanation somewhere here, on why is epoch needed and what's it all about? The importance of this would increase more as we have more contributors.

Reply:
There is a readme https://github.com/timescale/promscale/blob/master/pkg/pgmodel/epochs.readme.md but maybe we need to add more details to it

Reply:
^ what blagoj said